pkg/stub/handler.go

```diff
@@ -2,96 +2,401 @@ package stub
+					// testing showed that we get an empty is vs. nil in this case
+					is = nil
+				}
+
+				if is == nil {
+					_, err = h.imageclientwrapper.Create("openshift", imagestream)
+					if err != nil {
+						return h.processError(v1alpha1.SamplesUpdateFailed, err, "imagestream create error: %v")
+					}
+					logrus.Printf("created imagestream %s", imagestream.Name)
+				} else {
+					imagestream.ResourceVersion = is.ResourceVersion
+					_, err = h.imageclientwrapper.Update("openshift", imagestream)
+					if err != nil {
+						return h.processError(v1alpha1.SamplesUpdateFailed, err, "imagestream update error: %v")
+					}
+					logrus.Printf("updated imagestream %s", is.Name)
+				}
+			}
+		}
+
+		if strings.HasSuffix(dir, "templates") {
+			template, err := h.filetemplategetter.Get(dir + "/" + file.Name())
+			if err != nil {
+				return err
+			}
+
+			if _, tok := h.skippedTemplates[template.Name]; !tok {
+				t, err := h.templateclientwrapper.Get("openshift", template.Name, metav1.GetOptions{})
+				if err != nil && !kerrors.IsNotFound(err) {
+					return h.processError(v1alpha1.SamplesUpdateFailed, err, "unexpected template get error: %v")
+				}
+
+				if kerrors.IsNotFound(err) {
+					// testing showed that we get an empty is vs. nil in this case
+					t = nil
+				}
+
+				if t == nil {
+					_, err = h.templateclientwrapper.Create("openshift", template)
+					if err != nil {
+						return h.processError(v1alpha1.SamplesUpdateFailed, err, "template create error: %v")
+					}
+					logrus.Printf("created template %s", template.Name)
+				} else {
+					template.ResourceVersion = t.ResourceVersion
+					_, err = h.templateclientwrapper.Update("openshift", template)
+					if err != nil {
+						return h.processError(v1alpha1.SamplesUpdateFailed, err, "template update error: %v")
+					}
+					logrus.Printf("updated template %s", t.Name)
+				}
+			}
+		}
 	}
 	return nil
 }
 
-// newbusyBoxPod demonstrates how to create a busybox pod
-/*func newbusyBoxPod(cr *v1alpha1.SamplesResource) *corev1.Pod {
-	labels := map[string]string{
-		"app": "busy-box",
-	}
-	return &corev1.Pod{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Pod",
-			APIVersion: "v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "busy-box",
-			Namespace: cr.Namespace,
-			OwnerReferences: []metav1.OwnerReference{
-				*metav1.NewControllerRef(cr, schema.GroupVersionKind{
-					Group:   v1alpha1.SchemeGroupVersion.Group,
-					Version: v1alpha1.SchemeGroupVersion.Version,
-					Kind:    "SamplesResource",
-				}),
-			},
-			Labels: labels,
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Name:    "busybox",
-					Image:   "busybox",
-					Command: []string{"sleep", "3600"},
-				},
-			},
-		},
-	}
-}*/
+func (h *Handler) updateDockerPullSpec(old string, imagestream *imagev1.ImageStream) {
+	if len(h.operatorCfg.Spec.SamplesRegistry) > 0 {
+		if strings.HasPrefix(imagestream.Spec.DockerImageRepository, "docker.io") {
```

Comment:
all these docker.io constants should be "old" instead, right?

Reply:
yep, and I am about the push unit tests that validate this path as well

also had to change this function's algorithm for proper use of `strings.Replace`, as well as better deal with the replacing of two possible prefixes in the RHEL registry case.

Will push shortly.